adb_graphics/datahandler/grib.py

```diff
@@ -451,6 +451,36 @@ def grid_info(self):
 
         return grid_info
 
+    def supercooled_liquid_water(self, values, **kwargs) -> np.ndarray:
+
+        # pylint: disable=unused-argument
+
+        '''
+        Generates a field of Supercooled Liquid Water
+        '''
+
+        ps = self.values(name='pres', level='sfc') * 100. # convert back to Pa
+        p = self.values(name='pres', level='ua', one_lev=False)
+        t = self.values(name='temp', level='ua', one_lev=False)
+        cmr = self.values(name='clwmr', level='ua', one_lev=False)
+        rmr = self.values(name='rwmr', level='ua', one_lev=False)
+
+        g = 9.81 # gravity
+        slw = ps * 0.
+
+        nlevs = np.shape(p)[0]
```

Comment:
Maybe here a comment on the algorithm. Maybe that's something like:

>  Compute pressure on native levels by iteratively adding the average change in pressure over the layer to the known pressure at the next lower level, starting at the surface.

Reply:
added comments.